### PR TITLE
feat: add theme feature on Linux, closes #387

### LINKED ISSUE #387: [feat] Read dark/light theme on MacOS and Linux
**Describe the problem**

Currently, the API can only read out the current system theme on Windows, but not on MacOS.

The following code below will result in `light` even though I'm in dark mode.

```js
import { appWindow } from '@tauri-apps/api/window';

appWindow.theme().then((theme) => {
  console.log(theme); // returns 'light' even though I'm in dark mode
});
```

**Describe the solution you'd like**

This should work on MacOS as well.

**Alternatives considered**

None.

**Additional context**

N/A

### PR DESCRIPTION
**Pull request by keiya01**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [x] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [x] I have added a convincing reason for adding this feature, if necessary

### Other information
Currently GTK does not support theme changed event...

### CODE REVIEW

File: src/event.rs [hunk 1]
```diff
@@ -469,7 +469,9 @@ pub enum WindowEvent<'a> {
   /// Applications might wish to react to this to change the theme of the content of the window
   /// when the system changes the window theme.
   ///
-  /// At the moment this is only supported on Windows.
+  /// ## Platform-specific
+  ///
+  /// - **Linux / Android / iOS:** Unsupported
```
Comment (wusyong) [inline src/event.rs:474]: I assume it's Windows instead of Linux

Comment (keiya01) [reply src/event.rs:474]: I think windows is already supported.

**Review by wusyong — APPROVED** [review]

File: src/platform_impl/linux/window.rs [hunk 2]
```diff
@@ -647,6 +669,20 @@ impl Window {
...
     }
   }
+
+  pub fn theme(&self) -> Theme {
+    if let Some(settings) = Settings::default() {
+      let theme_name = settings.gtk_theme_name().map(|s| s.as_str().to_owned());
+      if let Some(theme) = theme_name {
+        // Currently GTK doesn't provide feature for detect theme, so we need to check theme manually.
+        // ref: https://github.com/WebKit/WebKit/blob/e44ffaa0d999a9807f76f1805943eea204cfdfbc/Source/WebKit/UIProcess/API/gtk/PageClientImpl.cpp#L587
+        if theme.ends_with("-dark") || theme.ends_with("-Dark") {
```
Comment (lucasfernog) [inline src/platform_impl/linux/window.rs:679]: I use Arc-Darker theme :( really sucks having to detect the theme this way @keiya01 @wusyong

Comment (amrbashir) [reply src/platform_impl/linux/window.rs:679]: probably should be
```suggestion
        if theme.contains("-dark") || theme.contains("-Dark") {
```

Comment (keiya01) [reply src/platform_impl/linux/window.rs:679]: Darker ... I see. I will fix this issue.

Comment (keiya01) [reply src/platform_impl/linux/window.rs:679]: @lucasfernog 
I created the PR. Could you check it when you have time?
https://github.com/tauri-apps/tao/pull/511

**Pull request merged**